src/numerics/petsc_matrix.C

```diff
@@ -1197,6 +1197,10 @@ void PetscMatrix<T>::matrix_matrix_mult (SparseMatrix<T> & X_in, SparseMatrix<T>
 
   ierr = MatMatMult(_mat, X->_mat, MAT_INITIAL_MATRIX, PETSC_DEFAULT, &Y->_mat);
   LIBMESH_CHKERR(ierr);
+
+  // Specify that the new matrix is initialized and close it.
+  Y->_is_initialized = true;
+  Y->close();
```

Comment:
You might just set closed flag to true. Do not need to really do communication. `_mat` from `MatMatMult` is closed for sure.

Reply:
Thanks @fdkong ! In this case, should I just remove `Y->close();` or do `libmesh_assert(Y->closed());` instead?

Reply:
Just remove `Y->close()`